OpenEdXMobile/src/main/java/org/edx/mobile/view/AuthenticatedWebViewFragment.java

```diff
@@ -0,0 +1,359 @@
+package org.edx.mobile.view;
+
+import android.annotation.SuppressLint;
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.StringRes;
+import android.support.v4.app.Fragment;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.JavascriptInterface;
+import android.webkit.WebResourceRequest;
+import android.webkit.WebResourceResponse;
+import android.webkit.WebView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import com.google.inject.Inject;
+import com.joanzapata.iconify.Icon;
+import com.joanzapata.iconify.IconDrawable;
+import com.joanzapata.iconify.fonts.FontAwesomeIcons;
+
+import org.edx.mobile.R;
+import org.edx.mobile.base.BaseFragment;
+import org.edx.mobile.event.NetworkConnectivityChangeEvent;
+import org.edx.mobile.event.SessionIdRefreshEvent;
+import org.edx.mobile.http.HttpStatus;
+import org.edx.mobile.logger.Logger;
+import org.edx.mobile.module.prefs.LoginPrefs;
+import org.edx.mobile.services.EdxCookieManager;
+import org.edx.mobile.util.NetworkUtil;
+import org.edx.mobile.view.custom.URLInterceptorWebViewClient;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import de.greenrobot.event.EventBus;
+import roboguice.inject.InjectView;
+
+import static org.edx.mobile.util.AppConstants.EMPTY_HTML;
+import static org.edx.mobile.view.Router.ARG_JAVASCRIPT;
+import static org.edx.mobile.view.Router.ARG_URL;
+
+public class AuthenticatedWebViewFragment extends BaseFragment {
```

Comment:
@miankhalid  The functionality you want to make common is not written atomic in `CourseUnitWebView`
so it needs to properly refactor `CourseUnitWebView` and separate out common reusable code as a Custom/Composite Webview and then use it here. It will take ~2 or 3 days.

cc: @verdan